lightning/src/ln/onion_utils.rs

```diff
@@ -974,14 +974,39 @@ where
 	const NODE: u16 = 0x2000;
 	const UPDATE: u16 = 0x1000;
 
+	enum ErrorHop {
+		RouteHop(RouteHop),
+	}
+
+	impl ErrorHop {
```

Comment:
Commit structure is so much better now. Much easier to review. Left a few comments about some splits into separate commits that might be a step too far.